xilem_web/src/concurrent/memoized_await.rs

```diff
@@ -1,16 +1,30 @@
 // Copyright 2024 the Xilem Authors and the Druid Authors
 // SPDX-License-Identifier: Apache-2.0
 
+use std::{fmt, future::Future, marker::PhantomData};
+
+use futures::{Stream, StreamExt};
+use wasm_bindgen::{closure::Closure, JsCast, UnwrapThrowExt};
+use wasm_bindgen_futures::spawn_local;
+
 use crate::{
     core::{MessageResult, Mut, NoElement, View, ViewId, ViewMarker, ViewPathTracker},
     DynMessage, OptionalAction, ViewCtx,
 };
-use std::{future::Future, marker::PhantomData};
-use wasm_bindgen::{closure::Closure, JsCast, UnwrapThrowExt};
-use wasm_bindgen_futures::spawn_local;
 
-/// Await a future returned by `init_future` invoked with the argument `data`, `callback` is called with the output of the future. `init_future` will be invoked again, when `data` changes. Use [`memoized_await`] for construction of this [`View`]
-pub struct MemoizedAwait<State, Action, OA, InitFuture, Data, Callback, F, FOut> {
+/// Await a future returned by `init_future` invoked with the argument `data`, `callback` is called with the output of the future.
+/// `init_future` will be invoked again, when `data` changes. Use [`memoized_await`] for construction of this [`View`]
+pub struct MemoizedAwait<State, Action, OA, InitFuture, Data, Callback, F, FOut>(
+    MemoizedInner<State, Action, OA, InitFuture, Data, Callback, F, FOut>,
+);
+
+/// Await a stream returned by `init_stream` invoked with the argument `data`, `callback` is called with the items of the stream.
+/// `init_stream` will be invoked again, when `data` changes. Use [`memoized_stream`] for construction of this [`View`]
+pub struct MemoizedStream<State, Action, OA, InitStream, Data, Callback, F, StreamItem>(
+    MemoizedInner<State, Action, OA, InitStream, Data, Callback, F, StreamItem>,
+);
+
+struct MemoizedInner<State, Action, OA, InitFuture, Data, Callback, F, FOut> {
```

Comment:
Maybe:

```suggestion
struct MemoizedFuture<State, Action, OA, InitFuture, Data, Callback, F, FOut> {
```

(but no strong feelings of the name, there are probably also better ones)